Meadowlark-js/tests/e2e/SchemaValidation.test.ts

```diff
@@ -0,0 +1,215 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+import { getAccessToken, Clients } from './functions/Credentials';
+import { baseURLRequest } from './Setup';
+
+describe('When posting a resource', () => {
+  describe('given a missing property on a required collection', () => {
+    it('should fail with message about the missing property', async () => {
+      // This is entirely missing the "categories" collection
+      await baseURLRequest
+        .post('/v3.3b/ed-fi/communityOrganizations')
+        .auth(await getAccessToken(Clients.Vendor1), { type: 'bearer' })
+        .send({
+          communityOrganizationId: 19,
+          nameOfInstitution: 'Communities in Schools',
+          shortNameOfInstitution: 'CIS',
+          identificationCodes: [
+            {
+              educationOrganizationIdentificationSystemDescriptor:
+                'uri://ed-fi.org/EducationOrganizationIdentificationSystemDescriptor#SEA',
+              identificationCode: '19',
+            },
+          ],
+        })
+        .expect(400)
+        .then((response) => {
+          expect(response.body.message).toMatchInlineSnapshot(`
+            [
+              " must have required property 'categories'",
+            ]
+          `);
```

Comment:
@andonyns this is the first time I have used an InlineSnapshot with Jest in my own test. It was really easy - I just wrote expect(response.body.message).toMatchInlineSnapshot('');` and let the test fail. Then I let jest (via VS Code integration) update all of the snapshots in the test. 

Then I could review the messages and ensure they are what I actually expected.

When we improve the error messages, it will be simple to update all of these snapshots automatically, and again manually inspect to make sure they are what we want. In that case, if the update results in message we don't like, that means a) the code is actually outputing the wrong message, and b) we should manually correct the inline snapshot.

Reply:
I saw this on the system tests and I like this approach

Reply:
Yes this is very handy. Jest also provides an excellent diff when a snapshot fails, making it easy to see what went wrong.